### PR TITLE
remove duplicate then statement; collapse duplicate if statements

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarCopyActionSpec.groovy [hunk 1]
```diff
@@ -118,11 +118,7 @@ public class TarCopyActionSpec extends Specification {
         action.execute(new CopyActionProcessingStream() {
             public void process(CopyActionProcessingStreamAction action) {
                 for (FileCopyDetailsInternal f : files) {
-                    if (f.isDirectory()) {
-                        action.processFile(f);
-                    } else {
-                        action.processFile(f);
-                    }
+                    action.processFile(f);
```
Comment (grimreaper) [inline subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarCopyActionSpec.groovy]: a cursory inspection seems to indicate this is the correct fix but there is a good chance I missed the logic here and really the the first if statement should be changed.

Comment (eriwen) [inline subprojects/core/src/test/groovy/org/gradle/api/internal/file/archive/TarCopyActionSpec.groovy]: This logic is correct assuming that `f.isDirectory()` has no side effects. I checked all implementors of `FileCopyDetails.isDirectory()` and that seems to be correct.

**Comment by eriwen** [comment]
@grimreaper Thanks for the quick cleanup PR. I'm afraid I don't see a CLA from you yet. Would you please [sign one](http://gradle.org/contributor-license-agreement/) so we can move forward?

**Comment by grimreaper** [comment]
assigned

**Comment by eriwen** [comment]
@grimreaper thanks! — merged

**Pull request merged**